providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/process/connector/grpc_watcher.py

```diff
@@ -153,7 +164,11 @@ def listen(self) -> None:
             if self.streamline_deadline_seconds > 0
             else {}
         )
-        request_args = {"selector": self.selector} if self.selector is not None else {}
+        request_args = {}
+        if self.selector is not None:
+            request_args["selector"] = self.selector
+        if self.provider_id is not None:
```

Comment:
is an empty `selector` or `provider_id` a valid use case? if not you can also remove the `is not None` part.

Reply:
The user would have to set an empty one explicitly so I don't think we should silently ignore it, if empty?